GlotPress/GlotPress
# Update filter counts via javascript

## Problem

When a translation status is changed through the editor, the top filter counts do not change.
Example:
1. Currently there are 100 current translated strings and 0 fuzzy.
If the user clicks on the "Fuzzy" button to set the translation to fuzzy status, the top filter counts remain `Translated (100)` and `Fuzzy (0)`.
2. On saving a translation, if the ammount of current/waiting/untranslated changes, all the matching filter counts should be updated
3. On discarding warnings, if lasts only one warning and after discarding it changes to 0, the Warnings filter should also be decreased by -1.

Cases:
- [x] Set status in the editor: Update filter counts, old and new status
- [x] Save translation: Update warnings and translation status filters
- [x] Discard warnings: If changes to 0 warnings, decrease fitler count and move to next row.

The counts are kept i18n.
Until there isn't a proper function like number_format_i18n() for js, use Intl.NumberFormat based on WP user locale slug ('en', 'pt', etc.).
See https://github.com/WordPress/gutenberg/issues/22628

## Solution
Use JavaScript to increment (+1) the new status filter and decrease (-1) the old status filter.

## Testing Instructions
1. Open a translation set with many different translation status.
2. Change the status of a single translation in the Editor.
4. See the old and new top filters change its counts on status change success. 

Fixes #858
